src/Symfony/Bundle/FrameworkBundle/Command/RouterDebugCommand.php

```diff
@@ -124,24 +128,45 @@ protected function execute(InputInterface $input, OutputInterface $output): int
                 'show_aliases' => $input->getOption('show-aliases'),
                 'output' => $io,
                 'container' => $container,
+                'method' => $method,
             ]);
         }
 
         return 0;
     }
 
-    private function findRouteNameContaining(string $name, RouteCollection $routes): array
+    private function findRouteNameContaining(string $name, RouteCollection $routes, ?string $method): array
     {
         $foundRoutesNames = [];
         foreach ($routes as $routeName => $route) {
-            if (false !== stripos($routeName, $name)) {
+            if (
+                false !== stripos($routeName, $name)
+                && (null === $method || $this->isMethodValidForRoute($method, $route->getMethods()))
```

Comment:
Why not simply:
```suggestion
                && (null === $method || 'ANY' === $method || \in_array($method, $route->getMethods(), true))
```

(please note that I've added `ANY` as well).

Reply:
Thanks for your comment! But if I do that, the routes with the 'any' method will no longer be displayed

Example:

Before: ![image](https://github.com/user-attachments/assets/f32ea9a0-120f-4c0f-894c-8ed10ad99496)

After: ![Screenshot from 2025-03-11 09-39-52](https://github.com/user-attachments/assets/c2d21e8c-f281-4d6c-bb0a-6e2eb3244df9)

Reply:
What about:
```php
(null === $method || 'ANY' === $method || $route->getMethods() && \in_array($method, $route->getMethods(), true))
```

Reply:
Thanks for the comment, I will do this, which is similar:
`(null === $method || 'ANY' === $method || ! $route->getMethods() || \in_array($method, $route->getMethods(), true))`

Reply:
Although I think I will remove the 'any' case, since if you want any route, you shouldn't use the method option.

Reply:
If you think it's necessary to add $method === 'ANY', please let me know. I don't think it's necessary because if you want any method, you shouldn't use that new option.

Reply:
Alright, let's keep it like that then :slightly_smiling_face: